src/contrib/inference/dynamichmc.jl

```diff
@@ -109,8 +109,8 @@ function AbstractMCMC.step(
     Q, _ = DynamicHMC.mcmc_next_step(steps, state.cache)
 
     # Update the variables.
-    vi[spl] = Q.q
-    DynamicPPL.setlogp!!(vi, Q.ℓq)
+    vi = DynamicPPL.setindex!!(vi, Q.q, spl)
+    vi = DynamicPPL.setlogp!!(vi, Q.ℓq)
```

Comment:
Do we have to redefine the log-density function since `l.vi` is potentially a different object?

Reply:
Oooo good point! Though, in that case we should probably just do away with the `state.vi` and use `state.logdensity.varinfo`, no?

Reply:
Yes, I guess. It seems `state.vi` is not even used in the DynamicHMC calls above but only state.logendensity.varinfo implicitly. I assume currently the values in the state.logdensity.varinfo are not completely crucial though since in the DynamicHMC calls its values are updated anyway.